controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -155,6 +155,35 @@ func (r *StorageConsumerReconciler) reconcileEnabledPhases() (reconcile.Result,
 		if err := r.deleteOnboardingSecret(); err != nil {
 			return reconcile.Result{}, nil
 		}
+
+		storageCluster, err := util.GetStorageClusterInNamespace(r.ctx, r.Client, r.namespace)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+
+		// isCephFsAvailable - get the default filesystem, isRbdAvailable - list blockpools CR with count 1, isNfsAvailable -
+		blockPoolList := &rookCephv1.CephBlockPoolList{}
+		if err := r.Client.List(r.ctx, blockPoolList, client.InNamespace(r.namespace), client.Limit(1)); err != nil {
+			return reconcile.Result{}, err
+		}
+		isRbdAvailable := len(blockPoolList.Items) > 0
+
+		cephFs := &rookCephv1.CephFilesystem{}
+		cephFs.Name = util.GenerateNameForCephFilesystem(storageCluster.Name)
+		cephFs.Namespace = r.namespace
+		if err := r.get(cephFs); client.IgnoreNotFound(err) != nil {
+			return reconcile.Result{}, fmt.Errorf("failed to get CephFilesystem: %v", err)
+		}
+		isCephFsAvailable := cephFs.UID != ""
+
+		cephNFS := &rookCephv1.CephNFS{}
+		cephNFS.Name = util.GenerateNameForCephNFS(storageCluster)
+		cephNFS.Namespace = r.namespace
+		if err := r.get(cephNFS); client.IgnoreNotFound(err) != nil {
+			return reconcile.Result{}, fmt.Errorf("failed to get CephNFS: %v", err)
+		}
+		isNfsAvailable := cephNFS.UID != ""
+
```

Comment:
Can we move this code into a function? one that returns available services?
RBD, CephFS, NFS, MCG

Reply:
creating a funtion and moving it to the utils